src/mrpro/data/_KData.py

```diff
@@ -171,3 +171,89 @@ def reshape_acq_data(data):
             )
 
         return cls(kheader, kdata, ktraj)
+
+    def to(
+        self,
+        device: torch.device | str | None = None,
+        dtype: None | torch.dtype = None,
+        non_blocking: bool = False,
+        copy: bool = False,
+        memory_format: torch.memory_format = torch.preserve_format,
+    ) -> KData:
+        """Perform dtype and/or device conversion of trajectory and data.
+
+        Parameters
+        ----------
+        device
+            The destination device. Defaults to the current device.
+        dtype
+            Dtype of the k-space data, can only be torch.complex64 or torch.complex128.
+            The dtype of the trajectory (torch.float32 or torch.float64) is then inferred from this.
+        non_blocking
+            If True and the source is in pinned memory, the copy will be asynchronous with respect to the host.
+            Otherwise, the argument has no effect.
+        copy
+            If True a new Tensor is created even when the Tensor already matches the desired conversion.
+        memory_format
+            The desired memory format of returned Tensor.
+        """
+        # Only complex64 and complex128 supported for kdata.
+        # This will then lead to a trajectory of float32 and float64, respectively.
+        if dtype is None:
+            dtype_traj = None
+        elif dtype == torch.complex64:
+            dtype_traj = torch.float32
+        elif dtype == torch.complex128:
+            dtype_traj = torch.float64
+        else:
+            raise ValueError(f'dtype {dtype} not supported. Only torch.complex64 and torch.complex128 is supported.')
+
+        return KData(
+            header=self.header,
+            data=self.data.to(
+                device=device, dtype=dtype, non_blocking=non_blocking, copy=copy, memory_format=memory_format
+            ),  # type: ignore [call-overload]
```

Comment:
is # type: ignore [call-overload] actually  requiered here?

Reply:
yes it is needed. mypy is also configured to throw an error if a type ignore is not needed

Reply:
which argment is the issue? 
I try to get this fixed in pytorch, but to looked ok to me.
` def to(self, device: Optional[DeviceLikeType] = None, dtype: Optional[_dtype] = None, non_blocking: _bool = False, copy: _bool = False, *, memory_format: Optional[torch.memory_format] = None) -> Tensor:`
is the typehint in pytorch. 
what am I missing?

Reply:
ah, it got recently fixed in https://github.com/pytorch/pytorch/pull/113647